iscsidev/iscsi.go

```diff
@@ -109,14 +109,15 @@ func (dev *Device) StartInitator() error {
 	}
 
 	// Setup initiator
-	err = nil
 	for i := 0; i < RetryCounts; i++ {
 		err = iscsi.DiscoverTarget(localIP, dev.Target, ne)
```

Comment:
Also use `err :=` here to replace `err = nil` above.

Reply:
I don't understand why err needs to be redeclared again, could you explain?

Reply:
It's just common practice to keep the scope of the variable clear. For example, It's normally
```
if err := xxxx; err != nil {
}
```
So we know this `err`'s scope is only for the sentence.

I don't really recall why this piece of code is written in this way
```
err = xxx
if IsTargetDiscovered(){
}
logrus.Warnf(err)
```

I guess I was trying to prevent the case that the second discover action failed but somehow the previous discover action result succeed. That's why the success depends on if the target has been found or not, instead of the error code.

In fact, I was a bit worried about adding the check of `if err != nil` here because it will miss the case that `err is nil` still can result in `fail to discover the target`. But later there is a log line for both `iscsi.CLeanupScsiNodes` so I thought it might be fine.

Now think about it, I would prefer keep `logrus.Warnf` without the err check. Because in it's patched form, it looks like this
```
err = xxxx
if IsTargetDiscovered {
    break
}
if err != nil {
    log
}
```

Which makes the reader tempting to move `if err != nil` to a few lines above to join the `err = xxx`, which will result in the logic change of the code.

Now I think it's better to do 
```
-	err = nil
...
+    err := iscsi.DiscoverTarget(localIP, dev.Target, ne)
```

And don't touch the warning part.

Reply:
Thanks, that makes more sense.